opal_client/client.py

```diff
@@ -74,6 +76,11 @@ def __init__(
         else:
             self.opa_runner = False
 
+
+        custom_ssl_context = get_custom_ssl_context()
+        if opal_common_config.SELF_SIGNED_CERTIFICATES_ALLOWED and custom_ssl_context is not None:
+            logger.warning("WARNING: OPAL client is configured to trust self-signed certificates!")
```

Comment:
This might be a bit too harsh.
We should defiently log this - but perhaps the "WARNING" and "!" are a bit much - since the user has made a conscious choice to use this...